### DISCUSSION TITLE
How can I use nestjs cli plugin with nx?

### DISCUSSION THREAD

**Discussion by Puszkarek** [nrwl/nx#23105] [General]
I'm trying to implement `swagger` plugin using nest docs: https://docs.nestjs.com/openapi/cli-plugin#using-the-cli-plugin.

But I wasn't able to make it work in a NX monorepo

I already checked these issues:
- https://github.com/nrwl/nx/issues/4135
- https://github.com/nrwl/nx/issues/2147

But looks that none of the solutions are working with the latest NX version

Is it a bug or there's a way to implement `nest-cli.json` using nx?

**Comment by MilanKovacic** [comment]
First, install the @nestjs/swagger integration package:

`npm install --save @nestjs/swagger`

if you are using the default setup, in your nest project, there should be a webpack configuration file, similar to this:

```javascript
const { NxWebpackPlugin } = require('@nx/webpack');
const { join } = require('path');

module.exports = {
  output: {
    path: join(__dirname, '../../dist/apps/my-workspace'),
  },
  plugins: [
    new NxWebpackPlugin({
      target: 'node',
      compiler: 'tsc',
      main: './src/main.ts',
      tsConfig: './tsconfig.app.json',
      assets: ['./src/assets'],
      optimization: false,
      outputHashing: 'none',
      transformers: [{ name: '@nestjs/swagger/plugin' }], // <-- ADD the following line
    }),
  ],
};
```

Add the transformer to the list of typescript compiler plugins as highlighted above.
Suggestions in the other threads don't work out of the box, as nx 18 introduced a new webpack infrastructure style by default.

**Reply by Puszkarek** [reply]
Thanks, it worked

**Reply by jaysoo** [reply]
Thanks @MilanKovacic

We can improve the generated app by showing how to use `transformers` option to consume plugins.
